plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/rest/MavenServerService.java

```diff
@@ -166,49 +166,39 @@ public Response reimportDependencies(@ApiParam(value = "The paths to projects wh
     }
 
     @GET
-    @Path("pom/reconsile")
+    @Path("pom/reconcile")
     @Produces("application/json")
-    public List<Problem> reconsilePom(@QueryParam("pompath") String pomPath) {
+    public List<Problem> reconcilePom(@QueryParam("pompath") String pomPath) {
         VirtualFileEntry entry = null;
         List<Problem> result = new ArrayList<>();
         try {
             entry = cheProjectManager.getProjectsRoot().getChild(pomPath);
             if (entry == null) {
                 return result;
             }
+
             Model.readFrom(entry.getVirtualFile());
             org.eclipse.che.api.vfs.Path path = entry.getPath();
             String pomContent = entry.getVirtualFile().getContentAsString();
             MavenProject mavenProject =
                     mavenProjectManager.findMavenProject(ResourcesPlugin.getWorkspace().getRoot().getProject(path.getParent().toString()));
-            if (mavenProject != null) {
-                List<MavenProjectProblem> problems = mavenProject.getProblems();
-                int start = pomContent.indexOf("<project ") + 1;
-                int end = start + "<project ".length();
-                List<Problem> problemList = problems.stream().map(mavenProjectProblem -> {
-                    Problem problem = DtoFactory.newDto(Problem.class);
-                    problem.setError(true);
-                    problem.setSourceStart(start);
-                    problem.setSourceEnd(end);
-                    problem.setMessage(mavenProjectProblem.getDescription());
-                    return problem;
-                }).collect(Collectors.toList());
-
-                List<Problem> missedArtifacts =
-                        mavenProject.getDependencies().stream()
-                                    .filter(mavenArtifact -> !mavenArtifact.isResolved())
-                                    .map(artifact -> {
-                                        Problem problem = DtoFactory.newDto(Problem.class);
-                                        problem.setError(true);
-                                        problem.setSourceStart(start);
-                                        problem.setSourceEnd(end);
-                                        problem.setMessage("Dependency " + artifact.getDisplayString() + " not found.");
-                                        return problem;
-                                    }).collect(Collectors.toList());
-
-                result.addAll(missedArtifacts);
-                result.addAll(problemList);
+            if (mavenProject == null) {
+                return result;
             }
+
+            List<MavenProjectProblem> problems = mavenProject.getProblems();
+            int start = pomContent.indexOf("<project ") + 1;
+            int end = start + "<project ".length();
+
+            List<Problem> problemList = problems.stream().map(mavenProjectProblem -> {
+                Problem problem = DtoFactory.newDto(Problem.class);
```

Comment:
shouldn't use .with*** to chain dto ?

return DtoFactory.newDto(Problem.class).with...with...with